lib/improver/tests/weighted_blend/test_WeightedBlend.py

```diff
@@ -168,8 +219,61 @@ def test_weights_equal_list(self):
     def test_weights_equal_array(self):
         """Test it works with weights set to array (0.8, 0.2)."""
         coord = "time"
-        plugin = BasicWeightedAverage(coord)
+        plugin = WeightedBlend(coord)
+        weights = np.array([0.8, 0.2])
+        result = plugin.process(self.cube, weights)
+        expected_result_array = np.ones((2, 2))*1.2
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_percentiles_weights_equal_none(self):
+        """Test it works for percentiles with weights set to None."""
+        coord = "time"
+        plugin = WeightedBlend(coord)
+        weights = None
+        perc_cube = percentile_cube()
+        result = plugin.process(perc_cube, weights)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA1,
+                                           (6, 2, 2))
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_percentiles_weights_equal_list(self):
+        """Test it works for percentiles with weights [0.8, 0.2]."""
+        coord = "time"
+        plugin = WeightedBlend(coord)
         weights = np.array([0.8, 0.2])
+        perc_cube = percentile_cube()
+        result = plugin.process(perc_cube, weights)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA2,
+                                           (6, 2, 2))
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_blend_percentile_cube1(self):
+        """Test blending percentile cube works with equal weights."""
+        coord = "time"
+        weights = np.array([0.5, 0.5])
+        perc_cube = percentile_cube()
+        plugin = WeightedBlend(coord)
+        result = plugin.process(perc_cube, weights)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA1,
+                                           (6, 2, 2))
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
+    def test_blend_percentile_cube2(self):
+        """Test blending percentile cube works with unequal weights."""
+        coord = "time"
+        weights = np.array([0.8, 0.2])
+        perc_cube = percentile_cube()
+        plugin = WeightedBlend(coord)
+        result = plugin.process(perc_cube, weights)
+        expected_result_array = np.reshape(BLENDED_PERCENTILE_DATA2,
+                                           (6, 2, 2))
+        self.assertArrayAlmostEqual(result.data, expected_result_array)
+
```

Comment:
These were the two I copied from the other test file, to match the unit test conventions. I guess the only extra things they are testing is the passing in weights as a list rather than an array and passing in weights=None. Seeing as these things are tested in other tests above, I would be happy to remove them.